src/docfx/lib/markdown/MarkdownUtility.cs

```diff
@@ -32,7 +32,7 @@ internal static class MarkdownUtility
             { "Caution", "Caution" },
         };
 
-        public static (string html, MarkupResult result) Markup(string markdown, Document file, Context context)
+        public static (string html, MarkupResult result) Markup(string markdown, Document file, Context context, ResolveHref resolveHref)
```

Comment:
This could be simplified to `Markup(md, file, context, Action<Document> buildChild)`

Reply:
we don't want to introduce the build action to lower level component? 

---
In reply to: [191296672](https://github.com/dotnet/docfx/pull/2818#discussion_r191296672) [](ancestors = 191296672)

Reply:
For markdown, the lower level component is markdig and extensions. This is just a helper and it already depend on `Document`

Reply:
build action should belongs to build, it's strange to have a build action in markup method, right? the markup is to render markdown to html?

---
In reply to: [191297651](https://github.com/dotnet/docfx/pull/2818#discussion_r191297651) [](ancestors = 191297651)